juniper_official/System/system-processes-memory.rule

```diff
@@ -25,37 +25,36 @@ healthbot {
         description "Monitors system statistics i.e. CPU, memory, storage, processes & queues and notifies anomalies";
         synopsis "System statistics analyzer";
         rule check-process-memory {
-            description "Collects system process's memory utilization periodically and notifies anomalies when daemon memory usage exceed threshold";
-            synopsis "System processes memory utilization analyzer";
             /*
              * Monitors *all* the daemons memory utilization. Notifies via the
              * dashboard colors when the daemon memory utilization exceed by
              * {{process-memory-high-threshold}} for every collected metric
              * during a 5 minute time range (red). The color is set to yellow
-             * if we see daemon memory utilization exceed 
+             * if we see daemon memory utilization exceed
              * {{process-memory-low-threshold}} for every collected metric
              * during a 5 minutes time range.
-             * 
+             *
              * Use daemon name and pid as key for rule.
              */
             keys [ daemon pid ];
-            /*
-             * Function derives memory utilization in percent from KB and MB.
-             */
-            function isExceeding {
-                description "Function isExceeding returns 'True' if % of size used out of resident memory is not exceeding the threshold";
-                path system-processes.py;
-                method isExceeding;
-                argument res {
-                    mandatory;
-                }
-                argument size {
+            synopsis "System processes memory utilization analyzer";
+            description "Collects system process's memory utilization periodically and notifies anomalies when daemon memory usage exceed threshold";
+            function decimal_to_percent {
+                path generic_functions.py;
+                method decimal_to_percent;
+                argument denominator {
```

Comment:
According to your unit testing results numerator is before denominator. Please do changes accordingly

Reply:
Modified the rule.